README.md

```diff
@@ -44,3 +44,42 @@ Example,
 	client.Save("api/v1/tenants", models.NewTenant(TenantAttributes))
     # TenantAttributes is struct present in models/tenant.go
 ```
+
+## Caching Support ##
+
+The client supports optional caching for schema requests to improve performance in scenarios with frequent schema lookups.
+
+### Enabling Caching ###
+
+Caching is **disabled by default** for safety. Enable it using the `CacheEnabled` option:
+
+```golang
+import "github.com/ciscoecosystem/mso-go-client/client"
+
+// Enable caching
+msoClient := client.GetClient("URL", "Username",
+    client.Password("Password"),
+    client.Insecure(true),
+    client.CacheEnabled(true))
+```
+
+### Cache Operations ###
+
+Once caching is enabled, you can use the following methods:
+
+```golang
+// Fetch schema with caching support (automatically handles cache hits/misses)
+schema, err := msoClient.GetSchemaWithCache("schema-id")
```

Comment:
why not adjust the current GetViaURL, so that when caching is enabled it leverage caching for everything?

Reply:
I personally think we should have consistent behaviour on all the resources, but this is not a resource (schema) discussion in my opinion but a generic caching discussion. In this case I would say we should have something like a GetViaURLWithCache function that can be used for schema ( but also for other endpoints ) where it would access the caching mechanism.

In the provider we can then gradually change those resources.